src/lib/registration/widgets/registration_code.rb

```diff
@@ -41,30 +39,33 @@ def label
 
       # Initialize the widget with stored values
       def init
-        log.info("Already registered") if registered?
+        reg_code = options.reg_code.to_s
 
-        @options = Storage::InstallationOptions.instance
-        self.value = @options.custom_url
-        self.value = @options.reg_code if !@options.reg_code.to_s.empty?
+        self.value = reg_code.empty? ? options.custom_url || boot_url : options.reg_code
```

Comment:
rubocop is not fine with the first parenthesis, but second one could help readability so changing it